File: base/src/main/java/de/whitefrog/frogr/repository/DefaultModelRepository.java
package de.whitefrog.frogr.repository;

import de.whitefrog.frogr.model.Model;

/**
 * Will be used by {@link RepositoryFactory} method when no other repository was found. 
 */
public class DefaultModelRepository<T extends Model> extends BaseModelRepository<T> {
  public DefaultModelRepository(String modelName) {
    super(modelName);
  }

  @Override
  public Class<?> getModelClass() {
    if(modelClass == null) {
      modelClass = cache().getModel(getType());
      if(modelClass == null) modelClass = Model.class;
    }

    return modelClass;
  }
}

File: base/src/main/java/de/whitefrog/frogr/repository/DefaultModelRepository.kt
package de.whitefrog.frogr.repository

import de.whitefrog.frogr.model.Model

/**
 * Will be used by [RepositoryFactory] method when no other repository was found.
 */
class DefaultModelRepository<T : Model>(modelName: String) : BaseModelRepository<T>(modelName) {
  override fun getModelClass(): Class<*> {
    if (modelClass == null) {
      modelClass = cache().getModel(type)
      if (modelClass == null) modelClass = Model::class.java
    }

    return modelClass
  }
}
